x/profiles/keeper/grpc_query.go

```diff
@@ -96,12 +96,10 @@ func (k Keeper) ChainLinks(ctx context.Context, request *types.QueryChainLinksRe
 	// Get user chain links prefix store
 	linksPrefix := types.ChainLinksPrefix
 	switch {
-	case request.User != "" && request.ChainName != "" && request.Target != "":
-		linksPrefix = types.ChainLinksStoreKey(request.User, request.ChainName, request.Target)
 	case request.User != "" && request.ChainName != "":
-		linksPrefix = types.UserChainLinksChainPrefix(request.User, request.ChainName)
+		linksPrefix = types.ChainLinksStoreKey(request.User, request.ChainName, request.Target)
 	case request.User != "":
-		linksPrefix = types.UserChainLinksPrefix(request.User)
+		linksPrefix = types.UserChainLinksChainPrefix(request.User, request.ChainName)
```

Comment:
Improve the prefix that `types.UserChainLinksChainPrefix("user", "")` is equal to `types.UserChainLinkPrefix("user")`